ShowerShow/Control/UserStatisticsController.cs

```diff
@@ -0,0 +1,96 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+using Microsoft.Extensions.Logging;
+using Microsoft.OpenApi.Models;
+using Newtonsoft.Json;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using System;
+using User = ShowerShow.Models.User;
+using CreateUserDTO = ShowerShow.DTO.CreateUserDTO;
+using AutoMapper;
+using ShowerShow.Repository.Interface;
+using ShowerShow.DTO;
+using ShowerShow.Models;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using ShowerShow.Utils;
+using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Extensions;
+using ShowerShow.Authorization;
+using System.Security.Claims;
+using ShowerShow.Model;
+
+namespace ShowerShow.Controllers
+{
+    public class UserStatisticsController
+    {
+        private readonly ILogger<UserStatisticsController> _logger;
+        private IUserStatisticsService userStatisticsService;
+        private IUserService userService;
+
+        public UserStatisticsController(ILogger<UserStatisticsController> log, IUserService userService, IUserStatisticsService userStatisticsService)
+        {
+            _logger = log;
+            this.userService = userService;
+            this.userStatisticsService = userStatisticsService;
+        }
+        [Function("GetFriendsRanking")]
+        [OpenApiOperation(operationId: "GetFriendsRanking", tags: new[] { "User Statistics" })]
+        [ExampleAuth]
+        [OpenApiParameter(name: "UserId", In = ParameterLocation.Path, Required = true, Type = typeof(Guid), Description = "The User ID parameter")]
+        [OpenApiParameter(name: "limit", In = ParameterLocation.Path, Required = true, Type = typeof(int), Description = "The limit")]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Dictionary<Guid, double>), Description = "The OK response with the friends ranking" + ".")]
+        public async Task<HttpResponseData> GetFriendsRanking([HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = "user/{UserId:Guid}/friendRanking/{limit}")] HttpRequestData req, Guid UserId, int limit, FunctionContext functionContext)
```

Comment:
I'm not sure if its me but when i used  the parameters below i got the internal server error with the error stack below
![image](https://user-images.githubusercontent.com/54850459/198188696-47bb6318-d8e8-4654-bc8e-3564e4ddeee5.png)

![image](https://user-images.githubusercontent.com/54850459/198188662-dbf32ce9-d455-4ae3-b451-87451e4b5c14.png)

![image](https://user-images.githubusercontent.com/54850459/198188627-eba5e9a3-114c-4112-b4af-c64ef50790e3.png)